packages/mui-base/src/utils/useComponentRenderer.ts

```diff
@@ -79,7 +79,7 @@ export function useComponentRenderer<
 
   let resolvedRenderProp:
     | ComponentRenderFn<React.HTMLAttributes<any>, OwnerState>
-    | React.ReactElement;
+    | React.ReactElement<any>;
```

Comment:
? https://github.com/mui/material-ui/pull/43402

```suggestion
    | React.ReactElement<unknown>;
```

Reply:
We don't really care about the shape of the props here, especially that it's an internal hook and its type shouldn't be publicly visible. To constrain it a bit more, I changed it to `Record<string, unknown>`